### PR TITLE
[centraldashboard] Sync iframe href on hashchange too

### PR DESCRIPTION
**Pull request by Bobgy**
Fixes https://github.com/kubeflow/kubeflow/issues/4598

/assign @avdaredevil

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/kubeflow/4684)

### CODE REVIEW

File: components/centraldashboard/public/components/iframe-container.js [hunk 1]
```diff
@@ -51,14 +51,19 @@ export class IframeContainer extends PolymerElement {
...
+            const syncIframePage = () => {
                 const iframeLocation = iframe.contentWindow.location;
                 const newIframePage = iframeLocation.href.slice(
                     iframeLocation.origin.length);
                 if (this.page !== newIframePage) {
                     this.page = newIframePage;
                 }
-            });
+            };
+            iframe.contentDocument.addEventListener('click', syncIframePage);
```
Comment (avdaredevil) [inline components/centraldashboard/public/components/iframe-container.js]: ***nit:** small nit for succinctness*

```suggestion
            const {contentDocument} = iframe
            contentDocument.addEventListener('click', syncIframePage);
            contentDocument.addEventListener('hashchange', syncIframePage);
```

**Comment by avdaredevil** [comment]
/hold
For change

/approve

**Comment by Bobgy** [comment]
/hold cancel

Thanks for review!

**Comment by avdaredevil** [comment]
/lgtm

Thanks Yuan!

**Pull request merged**